src/components/screens/settings-screen.tsx

```diff
@@ -16,54 +16,95 @@ const SettingsScreen = () => {
+    } else {
+      grantSeveraOptInConsent();
+    }
   };
 
   const grantSeveraOptInConsent = async () => {
     setIsLoading(true);
     try {
+      if (!usersApi) {
+        setError(strings.error.fetchFailedFlextime);
+        return;
+      }
       if (!userProfile?.email || !userProfile?.id) {
         setError(strings.error.missingEmailOrId);
         return;
       }
 
       const response = await usersApi.updateUserAttribute({
-        updateUserAttributeRequest: { email: userProfile?.email },
-        id: userProfile?.id,
+        updateUserAttributeRequest: { email: userProfile.email },
+        id: userProfile.id,
         attributeName: "isSeveraOptIn"
       });
 
-      const severaUserId = response.updatedKeycloakAttributes
-        ? response.updatedKeycloakAttributes.severaUserId
-        : undefined;
+      const severaUserIdRaw = response?.updatedKeycloakAttributes?.severaUserId;
+      const severaUserId = Array.isArray(severaUserIdRaw) ? severaUserIdRaw[0] : severaUserIdRaw;
 
       setIsConsentGiven(Boolean(severaUserId));
 
       if (severaUserId) {
-        const updatedProfile = {
-          ...userProfile,
-          attributes: {
-            ...userProfile.attributes,
-            severaUserId
-          }
-        };
+        const updatedAttributes = {
+          ...(userProfile.attributes || {}),
+          severaUserId
+        } as Record<string, string[] | string | undefined>;
 
+        const updatedProfile = { ...userProfile, attributes: updatedAttributes };
         setUserProfile(updatedProfile);
         setUsers((prev) =>
-          prev.map((u) =>
-            u.id === userProfile.id ? { ...u, attributes: { ...u.attributes, severaUserId } } : u
-          )
+          prev.map((u) => (u.id === userProfile.id ? { ...u, attributes: updatedAttributes } : u))
         );
       }
     } catch (error) {
-      console.error("Error fetching consent:", error);
-      setError(`${strings.error.fetchFailedFlextime}, ${error}`);
+      setError(`${strings.error.fetchFailedFlextime}, ${String(error)}`);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const revokeSeveraOptIn = async () => {
+    setIsLoading(true);
+    try {
+      if (!usersApi) {
+        setError(strings.error.fetchFailedFlextime);
+        return;
+      }
+      if (!userProfile?.id) {
+        setError(strings.error.missingEmailOrId);
+        return;
+      }
+
+      await usersApi.removeSeveraOptIn({ userId: userProfile.id });
+
+      const updatedAttributes = { ...(userProfile.attributes || {}) } as Record<
+        string,
+        string[] | string | undefined
+      >;
+      delete updatedAttributes.severaUserId;
+      delete updatedAttributes.isSeveraOptIn;
+
+      const updatedProfile = { ...userProfile, attributes: updatedAttributes };
+      setUserProfile(updatedProfile);
+      setUsers((prev) =>
+        prev.map((u) => (u.id === userProfile.id ? { ...u, attributes: updatedAttributes } : u))
+      );
```

Comment:
As said in a previous comment if this is a state patch to reflect the keycloak updates without the user having to logout then it's fine, but I wonder if there is a cleaner way to just refetch the keycloak data...